### PR TITLE
Switch `Jetpack_PostImages::from_gallery()` to using `get_post_galleries_images()` 

### LINKED ISSUE #1468: Post Images: Rework `Jetpack_PostImages::from_gallery()` to use `get_post_galleries()`
I've been porting some of the image parsing code out of Jetpack in the last couple of weeks to potentially use for upcoming BuddyPress features. While extracting and tidying and rewriting significants amount of code, I noticed that `Jetpack_PostImages::from_gallery()` could be drastically simplified by using WP's `get_post_galleries()` or related nearby functions.

The only thing the WP function doesn't do is try to extract any wrapping anchor element's `href`, but perhaps you could try to submit something upstream if you need to keep the feature.

### PR DESCRIPTION
**Pull request by egill**
Removes the custom functionality, switches to using
get_post_galleries_images().

`get_post_galleries_images()` returns a multidimentional array of image
URLs for all the galleries it finds in a post.

Could have used `get_post_galleries()`, decided against it since it
returns gallery specific data which is not used in this case.

The output is exactly the same as before, except now we're using the
built in WP function get the images.

Fixed #1468

### CODE REVIEW

File: class.jetpack-post-images.php [hunk 1]
```diff
@@ -85,64 +85,28 @@ static function from_gallery( $post_id ) {
 		$images = array();
 
 		$post = get_post( $post_id );
-		if ( !empty( $post->post_password ) )
+		if ( ! empty( $post->post_password ) ) {
 			return $images;
-
-		if ( false === has_shortcode( $post->post_content, 'gallery' ) ) {
-			return false; // no gallery - bail
```
Comment (kraftbj) [inline class.jetpack-post-images.php]: Why did you remove the early bailout?

**Comment by egill** [comment]
The bailout is present in get_post_galleries() (https://core.trac.wordpress.org/browser/trunk/src/wp-includes/media.php#L3299) which is called by get_post_galleries_images(). 

The cost of removing it from there is one extra get_post call and wp_list_pluck.

Does laving the bailout as is make sense considering the rest of Jetpack?

**Comment by dereksmart** [comment]
:+1: to this change. Using core's method we don't need our own custom bailout.  

@egill I'd like to get this into 3.8 -- mind putting it through a rebase?

**Comment by egill** [comment]
@dereksmart sure thing (sorry for the late reply), I'll get it done before the weekend.
